api/mymonero_rest.md

```diff
@@ -0,0 +1,391 @@
+API.
+
+## Encoding Schemes
+### JSON
+JSON is the original and required encoding scheme used for the API. Binary
+values (public keys, hashes, etc) are sent as an ascii-hex string. Some
+integers, that may exceed 2^53, are sent as strings. This is to due to the
+limitation within Javascript where all integers are double floating point
+values.
+
+## Transport Layers
+### HTTP-REST
+When calling an API method, the client must use HTTP POST to a path with the
+method name. As an example, to invoke the `get_address_txs` method, the client
+sends a POST message to `/get_address_txs` where the body contains the JSON
+request object associated with that method name. If the requested method does
+not exist, a HTTP 404 "Not Found" error must be returned. If the request type
+is not POST, the server shall return a HTTP 405 "Method Not Allowed" error. If
+the server is unable to complete a request temporarily due to load, then the
+server shall return a HTTP 503 "Service Unavailable" error to indicate to the
+client that the request may be serviceable later.
+
+Servers must support the JSON encoding scheme. The client must send the HTTP
+field `Content-Type: application/json`; if the server is not provided that
+content type from the client then the server shall respond with a HTTP 415
+"Unsupported Media Type" error.
+
+This transport layer does not use HTTP authentication, and instead uses the
+`view_key` field for authorization. Documentation for a specific method will
+indicate whether `view_key` is required. When it is not necessary, anyone can
+invoke the method.
+
+## Schema
+The ascii name of the field is used as a key in JSON encoding. If a field has
+a `*` indicator, that means the field is optional. If `*` is used next to the
+type, the value can be "null" or a value valid according to the type.
+
+### Types
+**binary**
+
+A hex-ascii string in JSON. This is generally some irreducible cryptographic
+concept - a public key or hash.
+
+**base58-address**
+
+A standard Monero public address encoded as a string in JSON.
+
+**output** object
+
+Information needed to spend an output.
+
+|       Field      |        Type       |          Description          |
+|------------------|-------------------|-------------------------------|
+| tx_id            | uint64            | Index of tx in blockchain     |
+| amount           | uint64-string     | XMR value of output           |
+| index            | uint16            | Index within vout vector      |
+| global_index     | uint64-string     | Index within amount           |
+| rct              | binary            | Bytes of ringct data          |
+| tx_hash          | binary            | Bytes of tx hash              |
+| tx_prefix_hash   | binary            | Bytes of tx prefix hash       |
+| public_key       | binary            | Output public key             |
+| tx_pub_key       | binary            | Ephemeral ECDH key            |
```

Comment:
Isn't this perhaps meant to be "Transaction public key"? I imagine an "ephemeral ECDH key" to be the output public key derived using the CryptoNote stealth addressing scheme `P = Hs(a*R | i)*G + B`.